doc/Development_Documentation/23_Installation_and_Upgrade/03_System_Setup_and_Hosting/02_Nginx_Configuration.md

```diff
@@ -171,6 +176,279 @@ server {
 }
 ```
 
+### Production Environment
+
+The following configuration provides an approperiate base for a secure application hosting. It can be adapted to your setup and preferences. However it is primarily taking security into account. It is recommended to develop within a secured environment, too.
+
+```nginx
+# mime types are covered in nginx.conf by:
+# http {
+#   include       mime.types;
+# }
+
+upstream php-pimcore6 {
+    server unix:/var/run/php/pimcore.sock;
+}
+
+server {
+    listen 80;
+    listen [::]:80;
+
+    server_name pimcore.loc;
+
+    root /var/www/pimcore/web;
+
+    # We accept .well-known in case of acme challenge (e.g. letsencrypt)
+    # Everything else, however, is return hostname with / location
+    # A good reference as explaination can be found here:
+    # https://www.digitalocean.com/community/tutorials/understanding-nginx-server-and-location-block-selection-algorithms#matching-location-blocks
+    location ~* /\.well-known/ {
+      try_files $uri /;
+    }
+
+    # Please note that return is cheaper than redirect    
+    # See: https://www.nginx.com/resources/wiki/start/topics/tutorials/config_pitfalls/#taxing-rewrites
+    location / {
+       return 301 https://pimcore.loc$request_uri;
```

Comment:
```suggestion
       return 301 https://$host$request_uri;
```

Should be possible here https://nginx.org/en/docs/http/ngx_http_core_module.html#var_host